### ISSUE TITLE
Running local tests

### ISSUE THREAD

**Issue by jonocodes** [cypht-org/cypht#1011]
## 💬 Question

What is the test methodology in cypht? Asking because I am trying to run at least the unit tests locally and getting failures.

```
$ cd /tests/phpunit
$ phpunit
PHPUnit 11.1.0 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.17
Configuration: /home/jono/src/cypht/tests/phpunit/phpunit.xml

.....FFFF................FEF..FFFF.......

ERRORS!
Tests: 168, Assertions: 374, Errors: 3, Failures: 10, Warnings: 4, Deprecations: 3401.
```

I see tests being run in the .travis dir, but I would think devs would want to run tests locally while they are working, not just in CI. Perhaps adding some development notes to the read me with instructions on how to run tests would be helpful.

**Comment by marclaporte** [comment]
@josaphatim please advise

**Comment by josaphatim** [comment]
@jonocodes
please just make sure you have a database, database user and password with the value **cypht_test**

```
$config->set('db_host', '127.0.0.1');
$config->set('db_name', 'cypht_test');
$config->set('db_user', 'cypht_test');
$config->set('db_pass', 'cypht_test');
```

If not you can change setup_db function in **tests/phpunit/mocks.php** and also change Hm_Test_DB::test_build_dsn in tests/phpunit/db.php to your own values. You can add the port to db_host if not using 3306 for mysql

I personally run tests with `php vendor/phpunit/phpunit/phpunit --configuration tests/phpunit/phpunit.xml` from the root folder

**Comment by marclaporte** [comment]
@kroky thoughts?

**Comment by kroky** [comment]
.github is specific to the CI, let's keep it there and not move it to tests.

+1 for the rest - devs should just be able to run phpunit and even selenium to execute their tests locally. If any specific setup is required and can't be done automatically, we should include documentation in a readme how to run the tests.

**Comment by IrAlfred** [comment]
Related issue: https://github.com/cypht-org/cypht/issues/1710

**Comment by IrAlfred** [comment]
[https://github.com/cypht-org/cypht/wiki/Running-local-Tests-in-Cypht](https://github.com/cypht-org/cypht/wiki/Running-local-Tests-in-Cypht)

**Comment by marclaporte** [comment]
@jonocodes I look forward to your feedback :-)

**Comment by jonocodes** [comment]
Yes, this is very much what I had in mind. And great writeup @IrAlfred .
